### PR TITLE
Analytics for `receive`

### PR DESCRIPTION
**Pull request by pedroapfilho**
- feat(analytics): adding platform to payload
- feat(analytics): added types for send plan
- feat(analytics): adding receive events

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts [hunk 1]
```diff
@@ -442,8 +459,107 @@ const analyticsMiddleware = () => (store) => (next) => (action) => {
...
+
+        break
+      }
+
+      case AT.components.request.GET_NEXT_ADDRESS: {
+        const state = store.getState()
+        const accountType =
+          state.forms.requestCrypto.values.selectedAccount.type === 'CUSTODIAL'
+            ? 'TRADING'
+            : 'USERKEY'
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts:473]: ideally all of this we should move to enums/const

Comment (pedroapfilho) [reply packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts:473]: You're right! I'm converting these to enums

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts [hunk 2]
```diff
@@ -74,22 +79,24 @@ const analyticsMiddleware = () => (store) => (next) => (action) => {
             const origin = simpleBuyOriginDictionary(rawOrigin)
 
             analytics.push(AnalyticsKey.BUY_SELL_CLICKED, {
+              analyticsType: AnalyticsType.EVENT,
               id,
               nabuId,
               origin,
               originalTimestamp: getOriginalTimestamp(),
-              type: AnalyticsType.EVENT
+              type: 'BUY'
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts:87]: This should be `OrderType.BUY` enum

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts [hunk 3]
```diff
@@ -74,22 +79,24 @@ const analyticsMiddleware = () => (store) => (next) => (action) => {
...
+              analyticsType: AnalyticsType.EVENT,
               id,
               nabuId,
               originalTimestamp: getOriginalTimestamp(),
               path: pathname,
               referrer,
               search,
               title,
-              type: AnalyticsType.EVENT,
+              type: 'BUY',
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts:99]: This should be `OrderType.BUY` enum

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts [hunk 4]
```diff
@@ -305,13 +330,18 @@ const analyticsMiddleware = () => (store) => (next) => (action) => {
             const inputAmount = Number(state.form.swapAmount.values.cryptoAmount)
             const inputCurrency = state.form.initSwap.values.BASE.coin
             const inputType =
-              state.form.initSwap.values.BASE.type === 'CUSTODIAL' ? 'TRADING' : 'USERKEY'
+              state.form.initSwap.values.BASE.type === 'CUSTODIAL'
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts:333]: There is a branch that's not merged in yet where I created a `SwapBaseCounterTypes` enum for this. You may want to add a TODO or make a note to come back and change this later.

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts [hunk 5]
```diff
@@ -442,8 +486,107 @@ const analyticsMiddleware = () => (store) => (next) => (action) => {
...
+        analytics.push(AnalyticsKey.BUY_SELL_VIEWED, {
+          analyticsType: AnalyticsType.EVENT,
+          id,
+          nabuId,
+          originalTimestamp: getOriginalTimestamp(),
+          path: pathname,
+          referrer,
+          search,
+          title,
+          type: 'SELL',
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts:588]: enum it

File: packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts [hunk 6]
```diff
@@ -95,14 +108,54 @@ type EmailVerificationClicked = BasePayload & {
   // origin: 'SIGN_UP' | 'VERIFICATION'
 }
 
-type SignedInPayload = BasePayload & {
-  platform: 'WALLET'
+type ReceiveCurrencySelectedPayload = BasePayload & {
+  account_type: 'SAVINGS' | 'TRADING' | 'USERKEY'
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts:112]: You created the `AccountType` enum but aren't using it here?

**Review by blockdylanb — APPROVED** [review]

**Pull request merged**